src/main/java/org/vanilladb/core/storage/index/btree/BTreeIndex.java

```diff
@@ -36,7 +37,8 @@
 public class BTreeIndex extends Index {
 	
 	protected static enum SearchPurpose { READ, INSERT, DELETE };
-
+	private static ConcurrentHashMap<String, Boolean> fileEmptyCache = new ConcurrentHashMap<String, Boolean>();

```

Comment:
I think we may be able to move this cache to `FileMgr` by adding a new API `isFileEmpty` or `isNewFile`.

Reply:
I agree. I'll create an API in FileMgr.